src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -44,30 +60,124 @@ public RateLimitingMiddleware(RequestDelegate next, ILogger<RateLimitingMiddlewa
     /// <returns>A <see cref="Task"/> that completes when the request leaves.</returns>
     public async Task Invoke(HttpContext context)
     {
-        using var lease = await TryAcquireAsync(context);
-        if (lease.IsAcquired)
+        using var leaseContext = await TryAcquireAsync(context);
+        if (leaseContext.Lease.IsAcquired)
         {
             await _next(context);
         }
         else
         {
+            var thisRequestOnRejected = _defaultOnRejected;
```

Comment:
Would it be bad to just always run this before any policy-specific `OnRejected` methods? That way we could make it a method instead of a `Func`.